pydantic/fields.py

```diff
@@ -253,9 +253,8 @@ def _populate_sub_fields(self) -> None:  # noqa: C901 (ignore complexity)
         else:
             raise TypeError(f'Fields of type "{origin}" are not supported.')
 
-        if getattr(self.type_, '__origin__', None):
-            # type_ has been refined eg. as the type of a List and sub_fields needs to be populated
-            self.sub_fields = [self._create_sub_type(self.type_, '_' + self.name)]
+        # type_ has been refined eg. as the type of a List and sub_fields needs to be populated
```

Comment:
I've been working a lot on this part of the code in #803, I'm now sure this code will need more work.

But maybe it would be easier for me to fix it in #803.